### PR TITLE
Add obi.js (BE revision)

### PR DESCRIPTION
**Pull request by smiled0g**

### CODE REVIEW

**Comment by smiled0g** [comment]
@sorawit Added BE to choose from under `ObiInteger`

File: obi/obi.js/index.js [hunk 1]
```diff
@@ -1,27 +1,227 @@
...
+
+  constructor(schema) {
+    this.isSigned = schema[0] === 'i'
+    this.sizeInBytes = parseInt(schema.slice(1)) / 8
+  }
+
+  /* TODO: Switch to BE */
+  encode(value) {
+    return this.encodeLE(value)
+  }
```
Comment (tansawit) [inline obi/obi.js/index.js]: should this now be `encodeBE` + remove TODO?

File: obi/obi.js/index.js [hunk 2]
```diff
@@ -1,27 +1,227 @@
...
+
+  /* TODO: Switch to BE */
+  decode(buff) {
+    return this.decodeLE(buff)
+  }
```
Comment (tansawit) [inline obi/obi.js/index.js]: ditto

File: obi/obi.js/index.js [hunk 3]
```diff
@@ -1,27 +1,227 @@
...
+    return Buffer.from(
+      [...Array(this.sizeInBytes)]
+        .map(() => {
+          const byte = value % BigInt(1 << 8)
+          value /= BigInt(1 << 8)
+          return parseInt(byte)
+        })
+        .reverse(),
+    )
+  }
```
Comment (tansawit) [inline obi/obi.js/index.js]: simplify this to
```js
encodeBE(value) {
	return encodeLE(value).reverse()
}
```

File: obi/obi.js/test.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    ).equals(
+      obi.encodeOutput({
+        price: 9268300000000,
+        sources: [
+          { name: 'CoinGecko', time: 1590305341 },
+          { name: 'CryptoCompare', time: 1590305362 },
+        ],
+      }),
+    ),
+  )
```
Comment (tansawit) [inline obi/obi.js/test.js]: This test case will need to be updated when switching number encoding to Big Endian

File: obi/obi.js/test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  check('Encode example input', () =>
+    Buffer.from('0300000042544300ca9a3b00000000', 'hex').equals(
+      obi.encodeInput({ symbol: 'BTC', multiplier: BigInt('1000000000') }),
+    ),
+  )
+})()
```
Comment (tansawit) [inline obi/obi.js/test.js]: update this test case as well

**Review by tansawit — CHANGES_REQUESTED** [review]

File: obi/obi.js/package.json [hunk 6]
```diff
@@ -5,7 +5,12 @@
...
-  "dependencies": {
-    "bn.js": "^5.1.2"
+  "scripts": {
+    "test": "node test.js"
+  },
+  "prettier": {
+    "trailingComma": "all",
+    "semi": false,
+    "singleQuote": true
   }
```
Comment (tansawit) [inline obi/obi.js/package.json:15]: Should we remove `prettier` as the library doesn't need it?

**Review by tansawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**